import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -548,6 +553,55 @@ protected boolean hasSerialHost(Record record) {
         return false;
     }
 
+    protected boolean isOnlineAccordingTo338(Record record) {
+        // Does the RDA carrier indicate that this is online?
+        List carrierFields = record.getVariableFields("338");
+        if (carrierFields != null) {
+            Iterator carrierFieldsIter = carrierFields.iterator();
+            while (carrierFieldsIter.hasNext()) {
+                DataField carrierField = (DataField) carrierFieldsIter.next();
+                String desc = carrierField.getSubfield('a') == null ? "" : carrierField.getSubfield('a').getData();
```

Comment:
When getting values with a fallback like this, you're always calling getSubfield() twice. In PHP we have the null coalesce operator `??` which unfortunately doesn't seem to be available in java. To avoid this, the only other efficient option would be to implement some kind of function `getSubfieldOrDefault(char code, String default)` that stores the subfield in a variable and calls getData() if it is != 0. 

But maybe i'm exaggerating here, since it would only improve performance on tags with lots of subfields...